pandas/core/frame.py

```diff
@@ -374,8 +376,12 @@ def _constructor_expanddim(self):
     # ----------------------------------------------------------------------
     # Constructors
 
-    def __init__(self, data=None, index=None, columns=None, dtype=None,
-                 copy=False):
+    def __init__(self,
+                 data=None,
+                 index: ArrayLike = None,
```

Comment:
these are ``Optional[Iterable[IUnion[Index, Iterable[str]]]]``  can you create a type in _typing, call it Axes (and give it a comment)

Reply:
ok, thanks, this was a bit confusing.

Reply:
@jreback what is `IUnion`? Also I think I need to use `ABCIndexClass` in _typing.py instead of `Index`.

Reply:
Shouldn't this just be `Iterable[Hashable]`? Parametrizating the generic Iterable with str seems too strict given these can also hold integers, tuples and even in some cases custom objects